File: android/app/src/main/java/com/riyadelberkawy/callcycle/MyInCallService.java
package com.riyadelberkawy.callcycle;

import android.telecom.Call;
import android.telecom.InCallService;

public class MyInCallService extends InCallService {
    private static Call currentCall;

    @Override
    public void onCallAdded(Call call) {
        super.onCallAdded(call);
        MyInCallService.currentCall = call;
    }

    @Override
    public void onCallRemoved(Call call) {
        super.onCallRemoved(call);
        MyInCallService.currentCall = null;
    }

    public static Call getCurrentCall() {
        return currentCall;
    }
}

File: android/app/src/main/java/com/riyadelberkawy/callcycle/MyInCallService.kt

package com.riyadelberkawy.callcycle

import android.telecom.Call
import android.telecom.InCallService

class MyInCallService : InCallService() {
    companion object {
        private var currentCall: Call? = null
    }

    override fun onCallAdded(call: Call) {
        super.onCallAdded(call)
        currentCall = call
    }

    override fun onCallRemoved(call: Call) {
        super.onCallRemoved(call)
        if (currentCall == call) {
            currentCall = null
        }
    }

    fun getCurrentCall(): Call? {
        return currentCall  
    }
}

